bundles/core/org.openhab.core/src/main/java/org/openhab/core/types/UnDefType.java

```diff
@@ -34,4 +34,20 @@ public String format(String pattern) {
 		return String.format(pattern, this.toString());
 	}
 
+	/**
+	 * Returns the enum constant with the specified name.
+	 */
+	public static UnDefType parse(String name) {
+		if (name == null) {
+			throw new NullPointerException("Name is null");
+		}
+		if (UNDEF.toString().equals(name)) {
+			return UNDEF;
+		} else if (NULL.toString().equals(name)) {
+			return NULL;
+		}
+		throw new IllegalArgumentException("No enum constant "
+				+ UnDefType.class.getCanonicalName() + "." + name);
+	}

```

Comment:
Yes, this is a case where a core type's `toString()` result is not acceptable to the class's `valueOf` static method.  But I'm concerned that there would be wide ripple effects from this change, as some item classes list `UnDefType` ahead of other types in the class's `acceptedDataType` list, so other code elsewhere in openHAB could experience a breaking change.  Could a code change to fix the call from a Jsr223 script to
```
postUpdate("ITEM_NAME", UnDefType.UNDEF);
```
happen in the Jsr223 code?  Is the script language just unable to reference Java enums using that syntax?